x-pack/platform/plugins/shared/fleet/public/applications/integrations/sections/epm/screens/installed_integrations/components/installed_integration_action_menu.tsx

```diff
@@ -80,97 +83,124 @@ export const InstalledIntegrationsActionMenu: React.FunctionComponent<{
-          </EuiToolTip>
-        ) : (
-          <FormattedMessage
-            id="xpack.fleet.epmInstalledIntegrations.bulkUninstallButton"
-            defaultMessage={'Uninstall {count, plural, one {# integration} other {# integrations}}'}
-            values={{
-              count: selectedItems.length,
-            }}
-          />
-        )}
-      </EuiContextMenuItem>,
-      ...(enablePackageRollback
-        ? [
-            <EuiContextMenuItem
-              key="rollback"
-              icon="returnKey"
-              disabled={!hasRollbackableIntegrations || !licenseService.isEnterprise()}
-              onClick={openRollbackModal}
-            >
+          </EuiContextMenuItem>,
+          <EuiContextMenuItem
+            key="uninstall"
+            icon="trash"
+            disabled={hasUninstallableIntegrations}
+            onClick={openUninstallModal}
+          >
+            {hasUninstallableIntegrations ? (
+              <EuiToolTip
+                position="right"
+                content={
+                  <FormattedMessage
+                    id="xpack.fleet.epmInstalledIntegrations.uninstallDisabledTooltip"
+                    defaultMessage="Can't uninstall integrations that are attached to agent policies"
+                  />
+                }
+              >
+                <FormattedMessage
+                  id="xpack.fleet.epmInstalledIntegrations.bulkUninstallButton"
+                  defaultMessage={
+                    'Uninstall {count, plural, one {# integration} other {# integrations}}'
+                  }
+                  values={{
+                    count: selectedItems.length,
+                  }}
+                />
+              </EuiToolTip>
+            ) : (
               <FormattedMessage
-                id="xpack.fleet.epmInstalledIntegrations.bulkRollbackButton"
+                id="xpack.fleet.epmInstalledIntegrations.bulkUninstallButton"
                 defaultMessage={
-                  'Rollback {count, plural, one {# integration} other {# integrations}}'
+                  'Uninstall {count, plural, one {# integration} other {# integrations}}'
                 }
                 values={{
                   count: selectedItems.length,
                 }}
               />
-            </EuiContextMenuItem>,
-          ]
-        : []),
-    ];
+            )}
+          </EuiContextMenuItem>,
+          ...(enablePackageRollback
+            ? [
+                <EuiContextMenuItem
+                  key="rollback"
+                  icon="returnKey"
+                  disabled={!hasRollbackableIntegrations || !licenseService.isEnterprise()}
+                  onClick={openRollbackModal}
+                >
+                  <FormattedMessage
+                    id="xpack.fleet.epmInstalledIntegrations.bulkRollbackButton"
+                    defaultMessage={
+                      'Rollback {count, plural, one {# integration} other {# integrations}}'
```

Comment:
```suggestion
                      'Roll back {count, plural, one {# integration} other {# integrations}}'
```
Nit: this needs to be in 2 words to be a verb, otherwise it's a noun